plugins/PrivacyManager/API.php

```
@@ -400,6 +409,36 @@ public function executeDataPurge(
         }
     }
 
+    /**
+     * @internal
+     */
+    public function getComplianceStatus(string $idSite, string $complianceType): array
```

A question - if we return `?array` and return null instead of empty arrays, would it minimise the number of empty test responses to have to cater for without impacting the front-end? Just wondering what's better when the combination of input params is invalid.

---

I don't think we should change the interface for the sake of fewer test files, however we are currently silently failing and not really informing the user what is wrong, which is possibly a bad API.

I've changed the approach, so we do always return an array OR throw an exception, it's already common/consistent with other APIs.